### PR TITLE
fix: issues affecting code quality

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes

- Remove unnecessary calls to fmt.Sprint
- Fix check for empty string
- Fix unnecessary calls to Printf
- Combine multiple `append`s into a single call

### CODE REVIEW

File: viz/tap/api/handlers.go [hunk 1]
```diff
@@ -252,21 +252,7 @@ func handleAPIResourceList(w http.ResponseWriter, _ *http.Request, _ httprouter.
...
-				Verbs:      metav1.Verbs{"watch"},
-			})
-		resList.APIResources = append(resList.APIResources,
-			metav1.APIResource{
-				Name:       fmt.Sprintf("%s/tap", res.name),
-				Namespaced: res.namespaced,
-				Kind:       gvk.Kind,
-				Verbs:      metav1.Verbs{"watch"},
-			})
+		resList.APIResources = append(resList.APIResources, metav1.APIResource{Name: res.name, ShortNames: []string{res.shortname}, Namespaced: res.namespaced, Kind: gvk.Kind, Verbs: metav1.Verbs{"watch"}}, metav1.APIResource{Name: fmt.Sprintf("%s/tap", res.name), Namespaced: res.namespaced, Kind: gvk.Kind, Verbs: metav1.Verbs{"watch"}})
```
Comment (kleimkuhler) [inline viz/tap/api/handlers.go:255]: I find this harder to read than before so I'd prefer to keep separate `append`s here.

**Review by kleimkuhler — COMMENTED** [review]
Thanks @withshubh! I'd like to keep one of these changes as-is, but everything else looks good.

We're working on the CI failures in a separate issue so this should pass once we fix the other issue.

File: viz/tap/api/handlers.go (hunk 1 above)
Comment (withshubh) [reply viz/tap/api/handlers.go]: reverted :sparkles:

**Comment by kleimkuhler** [comment]
@withshubh Great! Could you try merging in the latest `main`? I think the issue with CI failing should be resolved if you do that..

**Comment by withshubh** [comment]
@kleimkuhler Done :sparkles:

**Review by kleimkuhler — APPROVED** [review]
Thanks @withshubh!

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**